server/src/main/java/org/opensearch/index/shard/RemoteStoreRefreshListener.java

```diff
@@ -253,9 +264,9 @@ private boolean syncSegments() {
                             )
                         );
                     }
-                    // Capture replication checkpoint before uploading the segments as upload can take some time and checkpoint can
-                    // move.
+
                     long lastRefreshedCheckpoint = ((InternalEngine) indexShard.getEngine()).lastRefreshedCheckpoint();
```

Comment:
@ashking94 I think we should use segmentInfo to get refreshCheckpoint so that we can remain consistent with the time we took segmentInfoSnapshot to upload? #
```suggestion
                    long lastRefreshedCheckpoint = Long.parseLong(segmentInfos.getUserData().get(LOCAL_CHECKPOINT_KEY));
```